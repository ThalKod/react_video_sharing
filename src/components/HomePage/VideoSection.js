import React from "react";

import VideoSingle from "components/HomePage/VideoSingle";

export default class VideoSection extends React.Component{

  renderVideoList = () => {
    const { recommendedVideos, featuredVideos, videos } = this.props;

    if(featuredVideos && featuredVideos.length > 0 ){
      return featuredVideos.map(video => {
        return <VideoSingle key={video._id} {...video} />
      })
    }

    if(recommendedVideos && recommendedVideos.length > 0){
      return recommendedVideos.map(video => {
        return <VideoSingle key={video._id} {...video} />
      })
    }

    return videos.map(video => {
      return <VideoSingle key={video._id} {...video} />
    })

  };

  onScroll = () => {
    const { getMoreVideos } = this.props;
    if (// TODO: refactor scrolling and implement a better infinite bottom scroll for this component. lets' keep this for this mvp...
        window.innerHeight + document.documentElement.scrollTop === document.documentElement.offsetHeight
    ) {
      getMoreVideos()
    }
  };

  componentDidMount = () => {
    const { scrollable } = this.props;
    if(!scrollable) return;
    window.addEventListener("scroll", this.onScroll, false);
  };

  componentWillUnmount = () => {
    const { scrollable } = this.props;
    if(!scrollable) return;
    window.removeEventListener("scroll", this.onScroll, false);
  };

  render() {
    const { type, header } = this.props;
    return (
        // TODO: make it more generics...
        <div className="content-block head-div">
          { header && <div className="cb-header">
                        <div className="row">
                          <div className="col-lg-10 col-sm-10 col-xs-8">
                            <ul className="list-inline">
                              <li><a href="/" className="color-active">{type}</a></li>
                            </ul>
                          </div>
                        </div>
                      </div> }
          <div className="cb-content videolist">
            <div className="row">
              {this.renderVideoList()}
            </div>
          </div>
        </div>
    )
  }
}

