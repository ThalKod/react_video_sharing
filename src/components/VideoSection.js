import React from "react";
import VideoList from "./VideoList";

class VideoSection extends React.Component{

  state = {
    categories: "Featured",
  };

  render() {
    const { categories } = this.state;
    return (
        <div className="content-block head-div">
          <div className="cb-header">
            <div className="row">
              <div className="col-lg-10 col-sm-10 col-xs-8">
                <ul className="list-inline">
                  <li><a href="/" className="color-active">{categories}</a></li>
                </ul>
              </div>
            </div>
          </div>
          <VideoList/>
          <VideoList/>
        </div>
    )
  }
}

export default VideoSection;
