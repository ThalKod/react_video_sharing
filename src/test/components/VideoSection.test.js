import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";


import VideoSection from "../../components/Home/VideoSection";

describe("VideoSection", () => {
  it("should render the component correctly", () => {
    const wrapper = shallow(<VideoSection/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  // TODO: add more test later
});
