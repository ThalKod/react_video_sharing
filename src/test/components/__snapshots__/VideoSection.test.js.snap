// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoSection should render the component correctly 1`] = `
<div
  className="content-block head-div"
>
  <div
    className="cb-content videolist"
  >
    <div
      className="row"
    >
      <SimpleNote
        text="No Videos found"
      />
    </div>
  </div>
</div>
`;
